src/main/scala/beam/agentsim/infrastructure/power/SitePowerManager.scala

```diff
@@ -0,0 +1,153 @@
+package beam.agentsim.infrastructure.power
+
+import beam.agentsim.infrastructure.ChargingNetwork
+import beam.agentsim.infrastructure.ChargingNetwork.{ChargingCycle, ChargingStation, ChargingVehicle}
+import beam.agentsim.infrastructure.ChargingNetworkManager.ChargingZone
+import beam.agentsim.infrastructure.charging.ChargingPointType
+import beam.router.skim.event
+import beam.sim.BeamServices
+import cats.Eval
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future, TimeoutException}
+
+class SitePowerManager(chargingNetworkMap: TrieMap[String, ChargingNetwork], beamServices: BeamServices)
+    extends LazyLogging {
+  import SitePowerManager._
+
+  private val cnmConfig = beamServices.beamConfig.beam.agentsim.chargingNetworkManager
+  private val tazSkimmer = beamServices.skims.taz_skimmer
+  private val allChargingStations = chargingNetworkMap.flatMap(_._2.lookupStations).toList.distinct
+  private val unlimitedPhysicalBounds = getUnlimitedPhysicalBounds(allChargingStations).value
+
+  /**
+    * Get required power for electrical vehicles
+    *
+    * @param tick current time
+    * @return power (in Kilo Watt) over planning horizon
+    */
+  def requiredPowerInKWOverNextPlanningHorizon(tick: Int): Map[ChargingStation, PowerInKW] = {
+    lazy val planningFuture = Future
+      .sequence(allChargingStations.map { station =>
```

Comment:
Since you call `sequence` this is mostly a stylistic change than anything else, but you could change from Futures to parallel collections and it would likely reduce the code footprint. The biggest thing is that parallel collections do not have a simple mechanism for timeouts. So, if that is needed then simply resolve this and move on ;)

Reply:
I think it is a good idea. thanks